qemu/tests/qmp_command.py

```diff
@@ -19,167 +21,127 @@ def run(test, params, env):
-                            if (cmp_str not in str(qmp_o[i]) and
-                                    cmp_str_b not in str(qmp_o[i]) and
-                                    cmp_s not in str(qmp_o[i])):
-                                msg += ("\nCan not find '%s', '%s' or '%s' in "
-                                        " QMP command output."
-                                        % (cmp_s, cmp_str_b, cmp_str))
-                                test.fail(msg)
-                        elif qmp_cmd == "query-balloon":
-                            if (int(val) * 1024 * 1024 != qmp_o[key] and
-                                    val not in str(qmp_o[key])):
-                                msg += ("\n'%s' is not in QMP command output"
-                                        % val)
-                                test.fail(msg)
-                        else:
-                            if (val not in str(qmp_o[i][key]) and
-                                    str(bool(int(val))) not in str(qmp_o[i][key])):
-                                msg += ("\n'%s' is not in QMP command output"
-                                        % val)
-                                test.fail(msg)
-        elif result_check == "m_in_q":
-            res = output.splitlines(True)
-            msg = "Key value from human monitor command is not in"
-            msg += "QMP command output.\nQMP command output: '%s'" % qmp_o
-            msg += "\nHuman monitor command output '%s'" % output
-            for i in range(len(res)):
-                params = res[i].rstrip().split()
-                for param in params:
-                    if param.rstrip() in exception_list:
-                        continue
-                    try:
-                        str_o = str(qmp_o.values())
-                    except AttributeError:
-                        str_o = qmp_o
-                    if param.rstrip() not in str(str_o):
-                        msg += "\nKey value is '%s'" % param.rstrip()
-                        test.fail(msg)
-        elif result_check == "m_format_q":
-            match_flag = True
-            for i in qmp_o:
-                if output is None:
-                    test.error("QMP output pattern is missing")
-                if re.match(output.strip(), str(i)) is None:
-                    match_flag = False
-            if not match_flag:
-                msg = "Output does not match the pattern: '%s'" % output
-                test.fail(msg)
+                    test.fail("QMP output does not contain the expect value.\n"
+                              "Missed expect value: '%s'\n"
+                              "Actual result: '%s'\n" % (o, qmp_o))
+        elif result_check == "not_contain":
+            for o in expect_o:
+                if isinstance(o, dict):
+                    for key, val in o.items():
+                        result = check_list(qmp_o, key, val)
+                        if result:
```

Comment:
Well, seems this part are almost same as `result_check == "contain":`, besides here `result` or `not result`. What about this:
```
alter = False
if result_check == "not_contain":
    alter = True
if result_check == "contain" or result_check == "not_contain":
    for o in expect_o:
        ...
        if result and alter:
            break
    ...
```

Reply:
No, this method can not work,  since `alter=False` for `contain`,  `result and alter` always get `False`, this is not the expect result.

Reply:
@yanan-fu Oh right, forget about it.